### PR TITLE
Brotli MyGet Experiment

### PR DESCRIPTION
**Pull request by Vedin**
This PR to share the status and issues with Brotli shipping.

What we have now: BrotliNative.nupkg which contains brotli native libraries for different platforms (created using .nuspec file)

What we need: Upload it to dotnet.myget and use in brotli corefxlab project. 

Also, if we use package instead of copy dll's to the folder, we shouldn't specify a platform via command line and can work with project as usual. (I create a temporary feed with this package https://www.myget.org/feed/brotli-runtime-test/package/nuget/BrotliNative . So everybody can launch it on the machine. 

@KrzysztofCwalina @weshaggard @shiftylogic @ahsonkhan  Wait for you advices and suggestions.

### CODE REVIEW

**Comment by Vedin** [comment]
So how we understand after last commit, we can use native libraries from nuget correctly.

**Comment by Vedin** [comment]
@dotnet-bot test Innerloop Windows_NT Debug Build and Test

**Comment by Vedin** [comment]
@dotnet-bot test 
Innerloop Ubuntu16.04 Debug Build and Test

**Comment by Vedin** [comment]
@shiftylogic Can you please check what's the problem with: 
Restoring packages for D:\j\workspace\windows_nt_de---4b678204\tests\System.Buffers.Experimental.Tests\System.Buffers.Experimental.Tests.csproj on Innerloop Windows_NT Debug Build and Test

**Comment by Vedin** [comment]
@dotnet-bot test 
Innerloop Windows_NT Debug Build and Test

File: external/Brotli.nuspec [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        <authors>Microsoft</authors>
+        <requireLicenseAcceptance>false</requireLicenseAcceptance>
+        <licenseUrl>https://github.com/dotnet/corefxlab/blob/master/LICENSE</licenseUrl>
+        <description>Brotli build package</description>
+        <tags>Brotli</tags>
+    </metadata>
+    <files>
+        <file src="Brotli\lib\windows\x64\*" target="runtimes\win-x64\native" />
+        <file src="Brotli\lib\windows\x86\*" target="runtimes\win-x86\native" />
+        <file src="Brotli\lib\osx\*" target="runtimes\osx\native" />
```
Comment (shiftylogic) [inline external/Brotli.nuspec]: Why is this entry duplicated?

**Comment by Vedin** [comment]
@dotnet-bot close

File: scripts/package.ps1 [hunk 2]
```diff
@@ -23,15 +23,22 @@ Write-Host "** Building all NuGet packages. **"
...
     Write-Host "Creating NuGet package for $file..."
     Invoke-Expression "$dotnetExePath pack $file -c $Configuration -o $packagesPath --include-symbols --version-suffix $BuildVersion"
-
+    
     if (!$?) {
         Write-Error "Failed to create NuGet package for project $file"
     }
 }
 
+Ensure-Nuget-Exists
```
Comment (shiftylogic) [inline scripts/package.ps1]: Add a Write-Host call with a message saying you are creating the Brotli native package.

File: scripts/package.ps1 [hunk 3]
```diff
@@ -23,15 +23,22 @@ Write-Host "** Building all NuGet packages. **"
 foreach ($file in [System.IO.Directory]::EnumerateFiles("$repoRoot\src", "System*.csproj", "AllDirectories")) {
     Write-Host "Creating NuGet package for $file..."
     Invoke-Expression "$dotnetExePath pack $file -c $Configuration -o $packagesPath --include-symbols --version-suffix $BuildVersion"
-
+
```
Comment (shiftylogic) [inline scripts/package.ps1]: nit: Space issue here.

File: NuGet.Config [hunk 4]
```diff
@@ -21,4 +21,4 @@
     <add key="nugetbuild" value="https://www.myget.org/F/nugetbuild/api/v3/index.json" />
     <add key="roslyn" value="https://dotnet.myget.org/F/roslyn/" />
   </packageSources>
-</configuration>
\ No newline at end of file
+</configuration>
```
Comment (shiftylogic) [inline NuGet.Config]: discard the changes to this file.

File: external/Brotli.nuspec (hunk 1 above)
Comment (Vedin) [inline external/Brotli.nuspec]: removed

File: scripts/package.ps1 [hunk 5]
```diff
@@ -29,9 +29,17 @@ foreach ($file in [System.IO.Directory]::EnumerateFiles("$repoRoot\src", "System
     }
 }
 
+Ensure-Nuget-Exists
+Write-Host "** Creating BrotliNative NuGet packages. **"
+$brotliExternalFile = "$repoRoot\external\BrotliNative.nuspec"
+Invoke-Expression "$nugetPath pack $brotliExternalFile -Version 0.0.1 -o $packagesPath"
```
Comment (joshfree) [inline scripts/package.ps1]: would it be possible to change package.ps1 to support any project that has a checked-in nuspec file (versus the auto-gen could that exists today)?

Comment (joshfree) [inline scripts/package.ps1]: or even just a general-purpose external/*.nuspec for other packages like this in the future

https://github.com/dotnet/corefxlab/blob/master/scripts/package.ps1#L32-L35

**Pull request merged**